presto-tests/src/main/java/com/facebook/presto/tests/statistics/StatisticsAssertion.java

```diff
@@ -0,0 +1,116 @@
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import static com.facebook.presto.sql.planner.optimizations.Predicates.alwaysTrue;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.MATCH;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.NO_BASELINE;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.NO_ESTIMATE;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.defaultTolerance;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertEquals;
+
+public class StatisticsAssertion
+{
+    private final DistributedQueryRunner runner;
+
+    public StatisticsAssertion(DistributedQueryRunner runner)
+    {
+        this.runner = requireNonNull(runner, "runner is null");
+    }
+
+    public Result check(@Language("SQL") String query)
+    {
+        return new Result(metricComparisons(query));
+    }
+
+    public List<MetricComparison> metricComparisons(@Language("SQL") String query)
+    {
+        String queryId = runner.executeWithQueryId(runner.getDefaultSession(), query).getQueryId();
+        QueryPlan queryPlan = runner.getQueryPlan(new QueryId(queryId));
+
+        StageInfo stageInfo = runner.getQueryInfo(new QueryId(queryId)).getOutputStage().get();
+        return new MetricComparator().getMetricComparisons(queryPlan, stageInfo);
+    }
+
+    public static class Result
+    {
+        public static final Predicate<PlanNode> IS_OUTPUT_NODE = OutputNode.class::isInstance;
+
+        private final List<MetricComparison> metricComparisons;
+
+        public Result(List<MetricComparison> metricComparisons)
+        {
+            checkArgument(!metricComparisons.isEmpty(), "No metric comparisons given");
+            this.metricComparisons = ImmutableList.copyOf(metricComparisons);
+        }
+
+        public Result matches()
+        {
+            return match(alwaysTrue(), Metric.OUTPUT_ROW_COUNT, defaultTolerance());
+        }
+
+        public Result outputHas(Metric metric, MetricComparisonStrategy strategy)
+        {
+            return match(IS_OUTPUT_NODE, metric, strategy);
+        }
+
+        private Result match(Predicate<PlanNode> planNodeFilter, Metric metric, MetricComparisonStrategy strategy)
+        {
+            return testMetrics(planNodeFilter, metric, metricComparison -> metricComparison.result(strategy) == MATCH);
+        }
+
+        private Result testMetrics(Predicate<PlanNode> planNodeFilter, Metric metric, Predicate<MetricComparison> assertCondition)
+        {
+            List<MetricComparison> differingMetricComparisons = metricComparisons.stream()
+                    .filter(metricComparison -> metricComparison.getMetric() == metric)
+                    .filter(metricComparison -> planNodeFilter.test(metricComparison.getPlanNode()))
```

Comment:
this method should raise an assertion error if there's not a result in either of the two lines above. In other words, metric comparisons should contain an element for all metrics, for all nodes.

Alternatively: maybe we should introduce a `MetricComparisons` class to guard this property?

I think this adds weight to the previous comment (remove `planNodeFilter`) as applying it only leaves us with the assumption, that there's a MetricComparison for all metrics, for the root node only)